### ISSUE TITLE
MajorityExample Fails

### ISSUE THREAD

**Issue by dathayer23** [Cognexa/cxflow-examples#13]
Running Windows 10 Anaconda Package Manager Python 3.6.2

Executed the following commands with no problems

pip3 install cxflow cxflow-tensorflow --upgrade
git clone https://github.com/Cognexa/cxflow-examples.git
cd cxflow-examples

thern executed 
cxflow train majority

The model failed when it began executing the main loop.

Here is the log output
2017-09-22 10:40:37.000532: INFO    @common         : Creating dataset
2017-09-22 10:40:50.000264: INFO    @common         : 	MajorityDataset created
2017-09-22 10:40:50.000264: INFO    @common         : Creating a model
2017-09-22 10:40:50.000418: INFO    @model          : 	Creating TF model on 0 GPU devices
2017-09-22 10:40:51.000055: INFO    @common         : 	MajorityNet created
2017-09-22 10:40:51.000060: INFO    @common         : Creating hooks
2017-09-22 10:40:51.000061: INFO    @common         : 	ComputeStats created
2017-09-22 10:40:51.000065: INFO    @common         : 	SaveBest created
2017-09-22 10:40:51.000066: INFO    @common         : 	LogVariables created
2017-09-22 10:40:51.000069: INFO    @common         : 	CatchSigint created
2017-09-22 10:40:51.000070: INFO    @common         : 	StopAfter created
2017-09-22 10:40:51.000076: INFO    @common         : Creating main loop
2017-09-22 10:40:51.000076: INFO    @common         : Running the main loop
2017-09-22 10:40:51.000079: ERROR   @util           : Running the main loop failed
2017-09-22 10:40:51.000087: ERROR   @util           : Timer was ended but not started.
Traceback (most recent call last):
  File "c:\programdata\anaconda3\envs\tensorflow\lib\site-packages\cxflow\cli\common.py", line 271, in run
    main_loop.run_training()
  File "c:\programdata\anaconda3\envs\tensorflow\lib\site-packages\cxflow\main_loop.py", line 248, in run_training
    self._try_run(training)
  File "c:\programdata\anaconda3\envs\tensorflow\lib\site-packages\cxflow\main_loop.py", line 196, in _try_run
    run_func()
  File "c:\programdata\anaconda3\envs\tensorflow\lib\site-packages\cxflow\main_loop.py", line 224, in training
    self._run_zeroth_epoch([MainLoop.TRAIN_STREAM] + self._extra_streams)
  File "c:\programdata\anaconda3\envs\tensorflow\lib\site-packages\cxflow\main_loop.py", line 175, in _run_zeroth_epoch
    self.evaluate_stream(stream)
  File "c:\programdata\anaconda3\envs\tensorflow\lib\site-packages\cxflow\main_loop.py", line 139, in evaluate_stream
    self._run_epoch(stream=stream, train=False)
  File "c:\programdata\anaconda3\envs\tensorflow\lib\site-packages\cxflow\main_loop.py", line 107, in _run_epoch
    for batch_input in stream:
  File "c:\programdata\anaconda3\envs\tensorflow\lib\site-packages\cxflow\datasets\stream_wrapper.py", line 230, in __next__
    batch = get_batch_fn()
  File "c:\programdata\anaconda3\envs\tensorflow\lib\site-packages\cxflow\utils\profile.py", line 48, in __exit__
    raise ValueError('Timer was ended but not started.')
ValueError: Timer was ended but not started.

Really would like to get cxflow working.

**Comment by dathayer23** [comment]
Tried ConvNet and it also fails same errors

**Comment by dathayer23** [comment]
idmb example appears to work I have completed one epoch of training and am working on second epoch.

**Comment by blazekadam** [comment]
Hi dathayer23, it looks like a problem with **cxflow** profiling routines. We officially do not support Windows yet I will try to track down the issue. Thanks for the report.

**Comment by blazekadam** [comment]
Issue replicated & confirmed.

**Comment by blazekadam** [comment]
**Quickfix:** add `LogProfile` hook to `majority/config.yaml`:
```
  - LogProfile
```

**Comment by blazekadam** [comment]
This issue (and several others related to Windows) is now resolved in **cxflow** windows branch. We ll eventually merge it to master and pypi package; until then, you can install **cxflow** with:
```
pip install git+https://github.com/Cognexa/cxflow.git@windows
```

**Comment by petrbel** [comment]
@dathayer23 @blazekadam I just merged Cognexa/cxflow#138 to `dev` branch, so it should work when cxflow installed by 

```
pip install git+https://github.com/Cognexa/cxflow.git
```

The changes will be merged to pypi soon.

**Comment by dathayer23** [comment]
Thyank you
